selfservice/hook/verification.go

```diff
@@ -47,7 +51,16 @@ func (e *Verifier) do(r *http.Request, i *identity.Identity) error {
 			continue
 		}
 
-		token := link.NewVerificationToken(address, e.r.Config(r.Context()).SelfServiceFlowVerificationRequestLifespan())
+		verificationFlow, err := verification.NewPostHookFlow(e.r.Config(r.Context()).SelfServiceFlowVerificationRequestLifespan(), f)
```

Comment:
This needs tests :)

Reply:
Sure does. Will add to the tests in `selfservice/hook/verification_test.go` and some unit tests for `NewPostHookFlow`.

Reply:
hey, I'm wanting to add some test cases to `selfservice/hook_test.TestVerifier`, and I wanted to get a sense of what we should test for here. Should we be testing that the verification flow gets persisted to the DB and that it's as expected?

If so, I'm wondering how we can go about achieving this.  It seems verification flows can only be looked up by UUID currently, which isn't available in the test. Should I bring a DB mock (eg https://github.com/DATA-DOG/go-sqlmock) into the picture?

Let me know your thoughts, and if you actually had a different set of tests/testing strategy in mind here.

Reply:
> hey, I'm wanting to add some test cases to selfservice/hook_test.TestVerifier, and I wanted to get a sense of what we should test for here. Should we be testing that the verification flow gets persisted to the DB and that it's as expected?

I think that's a good approach!

> If so, I'm wondering how we can go about achieving this. It seems verification flows can only be looked up by UUID currently, which isn't available in the test. Should I bring a DB mock (eg https://github.com/DATA-DOG/go-sqlmock) into the picture?

I don't think that's necessary, you can execute a SQL raw query to find the object using `registry.Persister().RawQuery().First(&model)`

Reply:
added